### PR TITLE
Update CI docs

### PR DESCRIPTION
**Pull request by DropD**
## Description

Update CI docs and remove old CIGR files.

Moved to be based on a rebased branch to keep the commit history more readable.

## Requirements

Before submitting this PR, please make sure:

- [X] You have run the code checks, tests and documentation build successfully
- [X] All fixes and all new functionality are tested and documentation is up to date
- [X] You looked at the [review checklist](https://github.com/GridTools/gt4py/blob/main/REVIEW_CHECKLIST.txt)

Additionally, if this PR contains code authored by new contributors:

- [X] All the authors are covered by a valid contributor assignment agreement,
signed by the employer if needed, provided to ETH Zurich
- [X] The names of all the new contributors have been added to an updated
version of the AUTHORS.rst file included in the PR

### CODE REVIEW

File: docs/development/CI/cscs-ci.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# CSCS-CI Configuration
+
+## Initial Setup
+
+Follow the steps in the [CSCS-CI documentation](https://gitlab.com/cscs-ci/ci-testing/containerised_ci_doc). You will require the help of someone who can register the project to be allowed to run CI on CSCS machines.
```
Comment (egparedes) [inline docs/development/CI/cscs-ci.md:5]: Optional suggestion:
```suggestion
Follow the steps in the [CSCS-CI documentation](https://gitlab.com/cscs-ci/ci-testing/containerised_ci_doc). As mentioned in the documentation, you will require the help of someone who can register the project to be allowed to run CI on CSCS machines.
```

File: docs/development/CI/cscs-ci.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Current Configuration
+
+The configuration can be viewed and changed on the [setup page](https://cicd-ext-mw.cscs.ch/ci/setup_ci) after logging in with the credentials for "cscs-ci setup for gt4py", which can be found on the internal credentials store in the GT4Py folder.
```
Comment (egparedes) [inline docs/development/CI/cscs-ci.md:9]: Optional suggestion:
```suggestion
The configuration can be viewed and changed on the [setup page](https://cicd-ext-mw.cscs.ch/ci/setup_ci) after logging in with the credentials for "cscs-ci setup for gt4py", which can be found on the internal CSCS credentials store in the GT4Py folder.
```

File: docs/development/CI/cscs-ci.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### Whitelisted Users
+
+The core GT4Py developers should be whitelisted. This means that pipelines run on their PRs automatically and they can run them on other contributor's pipelines by commenting "cscs-ci run [pipeline name]".
```
Comment (egparedes) [inline docs/development/CI/cscs-ci.md:13]: Just a question: what is the current state of the whitelist? Which developers are already there? If the current core developers are already in the whitelist I would rephrase it slightly to emphasize that "**new**" core developers should be added to the whitelist.

File: docs/development/CI/cscs-ci.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+### Pipeline
+
+There can be more than one pipeline, each with it's own entry point yaml file. By default there is only one (named "default") which allows omitting the pipeline name in "cscs-ci run" comments on PRs. Each pipeline can override global settings like who is allowed to run them.
+
+### Regenerating Notification Token
+
+Use the `gridtoolsjenkins` github user (credentials in the internal credentials store) to follow the steps outlined on the setup page (see above). You may be able to regenerate the existing token instead of creating a new one.
```
Comment (egparedes) [inline docs/development/CI/cscs-ci.md:21]:
```suggestion
Use the `gridtoolsjenkins` github user (credentials in the internal CSCS credentials store) to follow the steps outlined on the setup page (see above). You may be able to regenerate the existing token instead of creating a new one.
```

File: docs/development/CI/infrastructure.md [hunk 5]
```diff
@@ -4,20 +4,111 @@
...
+other   -[#ForestGreen,thickness=2]d-> nxt
+
+src_sto -[#DarkCyan,thickness=2]d-> sto
+src_cab -[#DarkGoldenRod,thickness=2]d-> sto
+cfg_wfl -[#Tomato,thickness=2]d-> sto
+pkg_cfg -[#ForestGreen,thickness=2]d-> sto
+
+@enduml
+-->
+![](workflows.svg)
```
Comment (egparedes) [inline docs/development/CI/infrastructure.md:92]: Just a comment: have you considered to use mermaid (https://mermaid.live)? We could embed the source diagram in the markdown file as you do here, but it would allow us to easily edit it online (and it probably also look better?

Comment (egparedes) [reply docs/development/CI/infrastructure.md:92]: The test boxes in the diagram still use the old `Test / Something` naming (with the slash in the middle)

File: docs/development/CI/infrastructure.md [hunk 6]
```diff
@@ -32,6 +123,7 @@ The testing workflows already use a matrix strategy to run the automated tests o
 
 ## Future improvements
 
+- Reenable code coverage workflows (potentially on CSCS-CI).
 - Split code quality: it might be better to run code quality tools separate for each project in the monorepo.
 - Split documentation: once there is proper HTML documentation generated for the projects, it might make sense to have that run as one job per project.
-- Template for tests: although there is a reusable workflow for the code coverage uploading, it probably make sense to reuse some of the workflow description for the tests as well.
\ No newline at end of file
+- Template for tests: although there is a reusable workflow for the code coverage uploading, it probably make sense to reuse some of the workflow description for the tests as well.
```
Comment (egparedes) [inline docs/development/CI/infrastructure.md:129]: Is this still up to date? I don't think the code coverage template is still valid or useful. I do agree that we could provide a general tempalte for tests.

File: docs/development/CI/infrastructure.md [hunk 7]
```diff
@@ -4,20 +4,111 @@
```
Comment (egparedes) [inline docs/development/CI/infrastructure.md:4]: I would add a short description mentioning that the current implementation of the CI workflows is based on running on the CI service runners the same task automating tools which work locally on developers' workstations like tox and pre-commit, so any test or code checking workflow can be run independently by any developer at any time, without relying on the CI service.

File: docs/development/CI/cscs-ci.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# CSCS-CI Configuration
+
+## Initial Setup
+
+Follow the steps in the [CSCS-CI documentation](https://gitlab.com/cscs-ci/ci-testing/containerised_ci_doc). You will require the help of someone who can register the project to be allowed to run CI on CSCS machines.
+
+## Current Configuration
+
+The configuration can be viewed and changed on the [setup page](https://cicd-ext-mw.cscs.ch/ci/setup_ci) after logging in with the credentials for "cscs-ci setup for gt4py", which can be found on the internal credentials store in the GT4Py folder.
+
```
Comment (egparedes) [inline docs/development/CI/cscs-ci.md:10]: Shouldn't we add some info about the current workflow file (`ci/cscs-ci.yml`)? At least a mention to the current location and the general structure (maybe even a link to basic information about gitlab workflows syntax).

**Review by egparedes — COMMENTED** [review]
I have some suggestions but it looks good overall.

File: docs/development/CI/cscs-ci.md (hunk 3 above)
Comment (DropD) [reply docs/development/CI/cscs-ci.md:13]: Good point. Right now it's everyone who is in the core gt4py developers group. I didn't want to put any information that will go out of sync very quickly. But adding "new" is a good solution.

File: docs/development/CI/infrastructure.md (hunk 5 above)
Comment (DropD) [reply docs/development/CI/infrastructure.md]: Check the results here: https://github.com/DropD/gt4py/blob/ci-docs/docs/development/CI/infrastructure.md. It seems to have trouble rendering it, although mermaid does so fine: [](https://mermaid.live/edit#pako:eNqdlu1O2zAUhm_FilQVJEArDNiCNAnKQEgDusG0H02FXNtJrTp25zi0VcW979itS5Jmakqr5sN-n9fH59iVFwFRlAVh0GotEJfchGiB2rFQUzLC2rTdK8n1K4PHtsQm11i00Zv9tlqRXCvRj1-RRPDBYornWX95G6DDw2_ob477XRgG_cyx4GY-iORSm2nywl5ZH34oU7kmLEPgJhNGl6TtjIJnlhn0_ZVFwaDEOQnB2ku6EAfLOJZor9v7vV8rlzPj5Q9sZgrKd621JGuv2sC2jFr0Gha8hpiMmaT1nplR3vPJKI0TVutoZdmq_38-JbUPGG22QjaK1jY50mal1rcgJnHyMo1Ff6r02C6BzeygktAXc7PRmVYb3QyWzcqMmE21u6OYi_eoDhCbEZFTLhN4wunE9u2B7v0tVoIyvY8w5JwqAt1HKXVvRxrS7Oxsfn3UhTGrU56MkxeIsT-BEtrUEyVjnpQjGpTnuUKqU8rMXLDVPrG8CP-MuGEHmdFqzMKhgBEuilK_fp22J_LUS-8Z5Xn6qMmI0w3CFsERt5AET1xpNZU10mFTqV05TnoJuzrFmst14NdYj7tzvMlAFldMr_fjrnv5fPf44KGu0vIGVhDTVyJnJdSvCIeCDgsPPasUG1USr9cHZAgLdqsZk15-ozRsKtdUYnx9mlBFDmbeqG671axpvXYswM7JX14Fl-MnZ9BBNZGvHA6nnJpReDyZXVSx449hJ02xKvgZFVO2faDTrfoqcYYqed4-yPnuyJcmSBX6imor2gTtfEKlXbU9wk5nZ-J4Z-JkO7HBrJdAYes2GOr0Y9jZx7DzRlhwEKRMp5hTOJstrEkUwD9cCmegEB4pi3EuTBRE8g2kODfqaS5JEBoNVQ_yCcWGXXOcaJwGYYxFBq2woeDYcL8877lj39s_ILp1lw)

Comment (DropD) [reply docs/development/CI/infrastructure.md]: As for better looking, the link styles provided by mermaid don't seem to work as well for this. Otherwise it seems just equivalent.

File: docs/development/CI/infrastructure.md [hunk 9]
```diff
@@ -1,23 +1,186 @@
...
+other   -[#ForestGreen,thickness=2]d-> nxt
+
+src_sto -[#DarkCyan,thickness=2]d-> sto
+src_cab -[#DarkGoldenRod,thickness=2]d-> sto
+cfg_wfl -[#Tomato,thickness=2]d-> sto
+pkg_cfg -[#ForestGreen,thickness=2]d-> sto
+
+@enduml
+-->
+![](workflows.svg)
```
Comment (egparedes) [inline docs/development/CI/infrastructure.md:164]: I personally like more the mermaid version because is inlined here and doesn't really require installing extra tools, but it's just a personal opinion. In any case, we need to drop one of them, so pick up your favorite diagram and delete the other one.

**Review by egparedes — COMMENTED** [review]
It looks good. Just keep one of the versions of the diagram (preferably the mermaid one :laughing: ) and we're ready to go.

File: docs/development/CI/infrastructure.md (hunk 9 above)
Comment (DropD) [reply docs/development/CI/infrastructure.md]: I will keep mermaid, after fixing the readability for dark mode github.

**Review by egparedes — APPROVED** [review]
It looks good.

**Comment by egparedes** [comment]
cscs-ci run
